File: pueblo-golf-tournament-api/Dto/TournamentPlayerScoreDto.cs
using pueblo_golf_tournament_api.Base;

namespace pueblo_golf_tournament_api.Dto
{
    public class TournamentPlayerScoreDto : BaseDto
    {
        public long TournamentId { get; set; }
        public long PlayerId { get; set; }
        public long TeamId { get; set; }
        public long RegistrationId { get; set; }
        public long TournamentHoleId { get; set; }
        public long PlayerTeeTimeScheduleId { get; set; }
        public long TeeTimeScheduleId { get; set; }
        public int ActualStrokes { get; set; }
        public int MolaveScore { get; set; }
        public long ScorerId { get; set; }
    }
}

File: pueblo_golf_tournament_mobile/lib/dto/model/tournament-player-score-dto.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:pueblo_golf_tournament_mobile/base/base-dto.dart';

part 'tournament-player-score-dto.g.dart';

@JsonSerializable()
class TournamentPlayerScoreDto extends BaseDto {
  final int tournamentId;
  final int playerId;
  final int teamId;
  final int registrationId;
  final int tournamentHoleId;
  final int playerTeeTimeScheduleId;
  final int teeTimeScheduleId;
  final int actualStrokes;
  final int molaveScore;
  final int scorerId;

  factory TournamentPlayerScoreDto.fromJson(Map<String, dynamic> json) =>
      _$TournamentPlayerScoreDtoFromJson(json);

  TournamentPlayerScoreDto(
      {required super.id,
      required this.tournamentId,
      required this.playerId,
      required this.teamId,
      required this.registrationId,
      required this.tournamentHoleId,
      required this.playerTeeTimeScheduleId,
      required this.teeTimeScheduleId,
      required this.actualStrokes,
      required this.molaveScore,
      required this.scorerId});

  Map<String, dynamic> toJson() => _$TournamentPlayerScoreDtoToJson(this);
}
